melodies_monet/plots/surfplots.py

```diff
@@ -917,7 +1075,30 @@ def make_boxplot(comb_bx, label_bx, ylabel = None, vmin = None, vmax = None, out
                                'markersize': 20.0}}
     sns.set_style("whitegrid")
     sns.set_style("ticks")
-    sns.boxplot(ax=ax,x="variable", y="value",data=pd.melt(comb_bx), **boxplot_kwargs)
+    len_combx = len(comb_bx.columns)
+    if len_combx ==  3:
+        data_obs = comb_bx[comb_bx.columns[0]].to_frame().rename({comb_bx.columns[0]:'Value'},axis=1)
+        data_model1 = comb_bx[comb_bx.columns[1]].to_frame().rename({comb_bx.columns[1]:'Value'},axis=1)
+        data_model2 = comb_bx[comb_bx.columns[2]].to_frame().rename({comb_bx.columns[2]:'Value'},axis=1)
+        data_obs['model'] = model_name_list[0]
+        data_model1['model'] = model_name_list[1]
+        data_model2['model'] = model_name_list[2]
```

Comment:
In theory there could be more than two models. It could be better to generalize these two `if` branches into one.

Reply:
now is loop